src/components/Calculator.jsx

```diff
@@ -1,69 +1,63 @@
 /* eslint-disable linebreak-style */
 /* eslint-disable react/no-array-index-key */
 /* eslint-disable arrow-body-style */
-/* eslint-disable react/prefer-stateless-function */
 import './Calculator.css';
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import ButtonWraper from './ButtonWraper';
 import Screen from './Screen';
 import Button from './Button';
 import calculate from '../logic/calculate';
 
-class Calculator extends Component {
-  constructor(props) {
-    super(props);
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-  }
+  const { total, next, operation } = state;
 
-  setNewState = (value) => {
-    const newState = calculate(this.state, value);
-    this.setState(newState);
-  }
+  const setNewState = (value) => {
+    const newState = calculate(state, value);
+    setState(newState);
+  };
 
-  handleClick = (e) => {
+  const handleClick = (e) => {
     const buttonName = e.target.textContent;
-    this.setNewState(buttonName);
-  }
+    setNewState(buttonName);
+  };
 
-  render() {
-    const btnValues = [
-      ['AC', '+/-', '%', '÷'],
-      [7, 8, 9, 'x'],
-      [4, 5, 6, '-'],
-      [1, 2, 3, '+'],
-      [0, '.', '='],
-    ];
-    const { total, next, operation } = this.state;
-    return (
-      <div className="wrapper">
-        <Screen value={operation == null
-          ? total || next || '0'
-          : `${total} ${operation} ${next == null ? ' ' : next}`}
-        />
-        <ButtonWraper>
-          {
+  const btnValues = [
+    ['AC', '+/-', '%', '÷'],
+    [7, 8, 9, 'x'],
+    [4, 5, 6, '-'],
+    [1, 2, 3, '+'],
+    [0, '.', '='],
+  ];
+  return (
+    <div className="wrapper">
+      <Screen value={operation == null
+        ? total || next || '0'
+        : `${total} ${operation} ${next == null ? ' ' : next}`}
+      />
+      <ButtonWraper>
+        {
             btnValues.flat().map((btn, i) => {
               return (
                 <Button
                   key={i}

```

Comment:
[OPTIONAL] It is not recommended to use indexes as keys in React. That it's because they might change over time and affect the performance of the app. [You can read more about it here.](https://reactjs.org/docs/lists-and-keys.html) I would recommend you to just use the value of btn here. If you have already considered this then ignore this comment.